utm/conversion.py

```diff
@@ -320,7 +321,11 @@ def latlon_to_zone_number(latitude, longitude):
         elif longitude < 42:
             return 37
 
-    return int((longitude + 180) / 6) + 1
+    zone_number = int((longitude + 180) / 6) + 1
+    # Handle edge case of longitude = 180
+    if zone_number > 60:
+        zone_number = 1
```

Comment:
I'm not entirely convinced by this yet. Wouldn't it be better to restrict `longitude` to valid values?

something like this:

```python
while longitude >= 180:
    longitude -= 360
```

this would cause `longitude = 180` to be converted to `longitude = -180` and in `int((longitude + 180) / 6) + 1` this would evaluate to:

```
int((-180 + 180) / 6) + 1)
int(0 / 6) + 1)
int(0 + 1)
int(1)
```

which matches the expected `zone_number = 1`

Reply:
I absolutely agree that normalizing longitude to be in a "calculable" range `[-180, 180)` instead of `[-180, 180]` would solve it nicely!

I avoided that mainly because such a change would also be applicable to `lat_lon_to_zone_number` (which currently does no validation at all, and will openly return zone numbers that are arbitrarily high, for arbitrarily high longitudes).

Which is to say: I think that's a good change but I'd like to apply it there as well, but I had no idea if that's going to break someone's workflow (somehow).

I'll go ahead and push that change though/update the tests, if that particular workflow feels not worth prioritizing (and I would happily agree)